internal/pkg/addon/package_test.go

```diff
@@ -23,12 +32,41 @@ func TestPackage(t *testing.T) {
 		bucket = "mockBucket"
 	)
 
+	lambdaZipHash := sha256.New()
+	indexZipHash := sha256.New()
+	indexFileHash := sha256.New()
```

Comment:
im dumb are these hash sums supposed to be compared somewhere 😅

Reply:
they are used to create the `S3Path`s [here](https://github.com/aws/copilot-cli/pull/3826/files#diff-0344a1783d235315005ac2a895dc6548066195e3312402fa9850f698072420f9R57-R59) and then those are compared in the mock functions!

Reply:
Oh I see what you mean!
We are validating that the right hashes are generated here: 
```
.Upload(bucket, indexZipS3Path /* ensure the right hash. */, gomock.Any())
```